### PR TITLE
NEW: ProfilerMarkers for measuring enabling/disabling of InputActions and for binding resolution.

### PR DESCRIPTION
**Pull request by surfnerd**
### Description

Exposing more timing data of the InputSystem by adding ProfilerMarkers to `InputAction.Enable()` and `InputActionMap.ResolveBindings()`.  This is necessary to track frame times in the performance test project I am working on. 

### Testing status & QA

I have run test projects against this code in Unity 6.000.0.27f1 and 2019.4.40f1.   

### Overall Product Risks

The risk of this change is very low given that they only add profiler information. 

- Complexity: 0
- Halo Effect: 0

### Comments to reviewers

To continue being accountable for performance, I am adding more profiler markers to parts of the code that previously did not have them.  These changes will be tracked in a separate performance test project. 

### Checklist

Before review:

- [x] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - JIRA ticket linked, example ([case %<ID>%](https://issuetracker.unity3d.com/product/unity/issues/guid/<ID>)). If it is a private issue, just add the case ID without a link.
    - Jira port for the next release set as "Resolved".
- [ ] ~Tests added/changed, if applicable.~
    - ~Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.~
    - ~Performance tests.~
    - ~Integration tests.~
- [ ] ~Docs for new/changed API's.~
    - ~Xmldoc cross references are set correctly.~
    - ~Added explanation how the API works.~
    - ~Usage code examples added.~
    - ~The manual is updated, if needed.~

During merge:

- [x] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

After merge:

- [ ] ~Create forward/backward port if needed. If you are blocked from creating a forward port now please add a task to ISX-1444.~

### CODE REVIEW

File: Assets/Tests/InputSystem/Plugins/UITests.cs [hunk 1]
```diff
@@ -1793,13 +1793,15 @@ public IEnumerator UI_CanReleaseAndPressTouchesOnSameFrame()
                 .Matches((UICallbackReceiver.Event e) => e.pointerData.pointerType == UIPointerType.Touch).And
                 .Matches((UICallbackReceiver.Event e) => e.pointerData.position == secondPosition));
 
+#if UNITY_2021_2_OR_NEWER
```
Comment (surfnerd) [inline Assets/Tests/InputSystem/Plugins/UITests.cs:1796]: `EventType.PointerMove` was wrapped in `#if UNITY_2021_2_OR_NEWER` in its declaration.

File: Assets/Tests/InputSystem/CorePerformanceTests.cs [hunk 2]
```diff
@@ -1113,7 +1113,9 @@ public void Performance_OptimizedControls_ReadingPose4kTimes(OptimizationTestTyp
         "InputSystem.onAfterUpdate",
         "PreUpdate.NewInputUpdate",
         "PreUpdate.InputForUIUpdate",
-        "FixedUpdate.NewInputFixedUpdate"
+        "FixedUpdate.NewInputFixedUpdate",
+        "InputAction.Enable",
+        "InputActionMap.ResolveBindings"
```
Comment (stefanunity) [inline Assets/Tests/InputSystem/CorePerformanceTests.cs:1119]: please also add "InputAction.Disable" here, thanks

**Review by stefanunity — APPROVED** [review]
Thanks for the help!

File: Assets/Tests/InputSystem/CorePerformanceTests.cs (hunk 2 above)
Comment (surfnerd) [reply Assets/Tests/InputSystem/CorePerformanceTests.cs:1119]: added in 539258b

**Review by ekcoh — APPROVED** [review]
Code changes LGTM. @surfnerd I noticed there are a lot of formatting diffs on this branch. Just wondering whether its corrections by formatting tool or accidental formatting diffs? I also noticed you got a CI error that seem to be a recent regression we have noticed also on other branches so likely unrelated to this addition.

**Comment by surfnerd** [comment]
> Code changes LGTM. @surfnerd I noticed there are a lot of formatting diffs on this branch. Just wondering whether its corrections by formatting tool or accidental formatting diffs? I also noticed you got a CI error that seem to be a recent regression we have noticed also on other branches so likely unrelated to this addition.

@ekcoh I think the formatting changes were due to the indentation from the `using` block.  If you turn off whitespace changes, the diff will be easier to parse.  If it's too noisy, I can tweak my changes. 

Yes, I've noticed the CI failure on windows in the `CanInstantiateActionMap` test.  I don't believe it's related to my changes.  If you're ok with this getting merged, then by all means.

**Comment by ekcoh** [comment]
@surfnerd Recommend updating this based on develop and resolving the conflict since the unreliable test have now also been temporarily disabled until fixed. If adressed I believe this can land.

**Comment by surfnerd** [comment]
> @surfnerd Recommend updating this based on develop and resolving the conflict since the unreliable test have now also been temporarily disabled until fixed. If adressed I believe this can land.

@ekcoh updated.  Thank you

**Pull request merged**